src/components/RegisterForm.jsx

```diff
@@ -1,4 +1,160 @@
-import React from 'react';
+/* eslint-disable class-methods-use-this */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Form, Button, Spinner } from 'react-bootstrap';
+import { registerUser, showErrors } from '../store/actions/registerActions';
+import validate from '../utils/registerFormValidator';
+import store from '../store/store';
+import { IS_LOADING, LOGIN } from '../store/actions/actionTypes';
 
-const RegisterForm = () => <div className="main-container">Register</div>;
+const { REACT_APP_FRONTEND_URL } = process.env;
+
+export class ConnectedRegisterForm extends Component {
+  constructor() {
+    super();
+    this.state = {
+      email: '',
+      username: '',
+      password1: '',
+      password2: '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  swapForm() {
+    store.dispatch({ type: LOGIN });
+  }
+
+  handleFocus(event) {
+    event.target.classList.remove('error');
+  }
+
+  handleChange(event) {
+    this.setState({ [event.target.id]: event.target.value });
```

Comment:
@ElMonstro, pick `name` from `target` instead of 'id' or use `ref` instaed. check out more [here](https://www.skptricks.com/2018/06/why-to-use-refs-instead-of-ids.html).